File: spring-boot-iot-ui/src/api/emergencyPlan.ts
import { request } from './request';
import { buildQueryString } from './query';
import type { ApiEnvelope, IdType } from '../types/api';

/**
 * 应急预案 API
 */

// 应急预案接口定义
export interface EmergencyPlan {
      id: IdType;
      planName: string;
      riskLevel: string;
      description: string;
      responseSteps: string;
      contactList: string;
      status: number;
      tenantId: IdType;
      createTime: string;
      updateTime: string;
      createBy: number;
      updateBy: number;
      deleted: number;
}

export interface EmergencyPlanPageResult {
      total: number;
      pageNum: number;
      pageSize: number;
      records: EmergencyPlan[];
}

// 获取预案列表
export const getPlanList = (params?: {
      planName?: string;
      riskLevel?: string;
      status?: number;
}): Promise<ApiEnvelope<EmergencyPlan[]>> => {
      const queryString = buildQueryString(params);
      const path = queryString ? `/api/emergency-plan/list?${queryString}` : '/api/emergency-plan/list';
      return request<EmergencyPlan[]>(path, { method: 'GET' });
};

// 分页获取预案列表
export const pagePlanList = (params?: {
      planName?: string;
      riskLevel?: string;
      status?: number;
      pageNum?: number;
      pageSize?: number;
}): Promise<ApiEnvelope<EmergencyPlanPageResult>> => {
      const queryString = buildQueryString(params);
      const path = queryString ? `/api/emergency-plan/page?${queryString}` : '/api/emergency-plan/page';
      return request<EmergencyPlanPageResult>(path, { method: 'GET' });
};

// 获取预案详情
export const getPlanById = (id: IdType): Promise<ApiEnvelope<EmergencyPlan>> => {
      return request<EmergencyPlan>(`/api/emergency-plan/get/${id}`, { method: 'GET' });
};

// 新增预案
export const addPlan = (data: Partial<EmergencyPlan>): Promise<ApiEnvelope<EmergencyPlan>> => {
      return request<EmergencyPlan>('/api/emergency-plan/add', { method: 'POST', body: data });
};

// 更新预案
export const updatePlan = (data: Partial<EmergencyPlan>): Promise<ApiEnvelope<EmergencyPlan>> => {
      return request<EmergencyPlan>('/api/emergency-plan/update', { method: 'POST', body: data });
};

// 删除预案
export const deletePlan = (id: IdType): Promise<ApiEnvelope<void>> => {
      return request<void>(`/api/emergency-plan/delete/${id}`, { method: 'POST' });
};

File: spring-boot-iot-alarm/src/main/java/com/ghlzm/iot/alarm/entity/EmergencyPlan.java
package com.ghlzm.iot.alarm.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 应急预案实体
 */
@Data
@TableName("emergency_plan")
public class EmergencyPlan implements Serializable {
      private static final long serialVersionUID = 1L;

      /**
       * 主键ID
       */
      @TableId(type = IdType.AUTO)
      private Long id;

      /**
       * 预案名称
       */
      private String planName;

      /**
       * 适用告警等级：red/orange/yellow/blue
       */
      private String alarmLevel;

      /**
       * 历史风险等级兼容字段
       */
      private String riskLevel;

      /**
       * 描述
       */
      private String description;

      /**
       * 响应步骤（JSON格式）
       */
      private String responseSteps;

      /**
       * 联系人列表（JSON格式）
       */
      private String contactList;

      /**
       * 状态：0-启用，1-停用
       */
      private Integer status;

      /**
       * 租户ID
       */
      private Long tenantId;

      /**
       * 创建时间
       */
      private LocalDateTime createTime;

      /**
       * 更新时间
       */
      private LocalDateTime updateTime;

      /**
       * 创建人
       */
      private Long createBy;

      /**
       * 更新人
       */
      private Long updateBy;

      /**
       * 逻辑删除：0-未删除，1-已删除
       */
      private Integer deleted;
}
